staging/src/k8s.io/component-base/logs/example/cmd/logger.go

```diff
@@ -28,26 +28,37 @@ import (
 )
 
 func main() {
+	err := runLoggerCommand()
+	defer func() {
+		if err != nil {
+			os.Exit(1)
+		}
+	}()
+}
+
+func runLoggerCommand() error {
 	command := NewLoggerCommand()
 	logs.InitLogs()
 	defer logs.FlushLogs()
 
 	if err := command.Execute(); err != nil {
-		os.Exit(1)
+		return err
 	}
+	return nil
 }
 
 func NewLoggerCommand() *cobra.Command {
 	o := logs.NewOptions()
 	cmd := &cobra.Command{
```

Comment:
IMO, it is better to add `SilenceErrors: true`. wdyt?

reason: https://github.com/kubernetes/kubernetes/pull/104503#discussion_r696769016

Reply:
I think so. done:)